### PR TITLE
Add Support for processing existing resources

### PR DESCRIPTION
**Pull request by iocanel**
The idea is to allow ap4k customize / decorated handcrafted resources, or resources generated by other tools, e.g. fabric8-maven-plugin or gradle-kubernetes-plugin.

See #22 for details.

### CODE REVIEW

**Comment by geoand** [comment]
One of the examples seems to be failing

**Comment by iocanel** [comment]
@geoand: the tests have been fixed now.

**Comment by geoand** [comment]
Cool I'll review this on Monday if you don't mind :)

**Comment by iocanel** [comment]
@geoand: I was hoping to release later today, so that I can also create a demo video. Maybe @cmoulliard: can review it today?

File: core/src/main/java/io/ap4k/processor/AbstractAnnotationProcessor.java [hunk 1]
```diff
@@ -26,29 +28,57 @@
...
 
 public abstract class AbstractAnnotationProcessor<C extends Configuration> extends AbstractProcessor {
 
   protected static final String PACKAGE = "";
-  protected static final String FILENAME = "META-INF/ap4k/%s.%s";
-  protected static final String CONFIG = "META-INF/ap4k/.config/%s.%s";
-  protected static final String PROJECT = "META-INF/ap4k/.project.%s";
+  protected static final String FILENAME = "%s.%s";
+  protected static final String CONFIG = ".config" + File.separatorChar + "%s.%s";
+  protected static final String PROJECT = "META-INF" + File.separatorChar + "ap4k" + File.separatorChar + ".project.%s";
```
Comment (geoand) [inline core/src/main/java/io/ap4k/processor/AbstractAnnotationProcessor.java]: These could probably be `Paths.get(...)`

**Comment by iocanel** [comment]
@geoand: Thanks. I will merge this and follow up with documentation right after.

**Comment by geoand** [comment]
Sounds good @iocanel !

**Pull request merged**